cmd/soci/commands/push.go

```diff
@@ -145,7 +148,41 @@ if they are available in the snapshotter's local content store.
 			} else {
 				dst.Client = authClient
 			}
+			existingIndexOption := cliContext.String(internal.ExistingIndexFlagName)
+			if ok := internal.ValidExistingIndexOptions[existingIndexOption]; !ok {
+				return fmt.Errorf("unexpected value for flag --existing-index: %s, expected types [%s, %s, %s, %s]",
```

Comment:
nit: the default formatting for a `[]string` is what you have here, maybe it would be easier to use: 
```
return fmt.Errorf("unexpected value for flag --existing-index: %s, expected types %v",
    existingIndexOption, []string{internal.Warn, internal.Skip, internal.Allow, internal.Abort})
```

Reply:
This should be defined together with those const definitions, e.g.,

in `cmd/soci/commands/internal/index.go`:

```golang
var SupportedExistingIndexOptions = []string{Warn, Skip, Allow, Abort}
```

Reply:
Yeah the reason I opted for a map was to have the ability to perform a check operation without having an extra helper loop through a slice eg: 
```
func checkSupported(v string) { 
found := false
for _,o := range SupportedExistingIndexOptions {
  if  o == v { found = true}
}
return found

```

Reply:
I think it's fine to have a slice and a helper `CheckSupported`, both in internal/index.go. Now we're exposing a function, instead of a struct, to be used by others, which I think it's better. And we can print the slice directly. So when you change anything (+/- an option), you don't need to make changes here.

Also containerd uses slice to check plugin capabilities (similar here) in its introspection service:

https://github.com/containerd/containerd/blob/759a6350bc16523c64639e0f75ef9dfc71a4832e/plugin/context.go#L69

https://github.com/containerd/containerd/blob/759a6350bc16523c64639e0f75ef9dfc71a4832e/api/services/introspection/v1/introspection.pb.go#L75